e2eAIOK/DeNas/search/MOSigoptSearchEngine.py

```diff
@@ -114,7 +114,7 @@ def search(self):
                 nas_score, score, latency = self.cand_evaluate(cand)
                 self.logger.info('epoch = {} structure = {} nas_score = {} params = {}'.format(epoch, cand, self.vis_dict[cand]['score'], self.vis_dict[cand]['params']))
                 metrics = []
-                metrics.append({'name': 'Score', 'value': score.item()})
+                metrics.append({'name': 'Score', 'value': score})
```

Comment:
@tianyil1 @zigzagcai ,  does score.item() cause mem leak?
I suppose without .item() only obj reference is used. And item() causes some memcpy?

Reply:
Because in the "the transformer_proxy.py" it has already been fixed to return the score.item(), where it was previously returned as a tensor and caused the memory leak issue, it did not need the ".item()" here.

Reply:
thanks, add explanation here
![image](https://user-images.githubusercontent.com/4355494/222640497-8c719193-16b1-4d93-8cd0-0bce1df71905.png)
root cause:
* original nas_score is a complete tensor object, which stored very large data
* in DeNas search, returned nas_score will be stored in searched dict to identify top candidate -> all big tensor object will be kept in memory.

Fix:
* only return nas_score value using `nas_score.item()` which is a float.

other lines is a code clean up